test/units/TEST-72-SYSUPDATE.sh

```diff
@@ -73,6 +74,17 @@ update_now() {
     (! "$SYSUPDATE" --definitions="$WORKDIR/defs" --verify=no check-new)
 }
 
+updatectl_update_now() {
+    # Update to the newest version using updatectl.
+
+    # Ensure sysupdated is running
+    systemctl start systemd-sysupdated
+
+    "$SYSUPDATE" --definitions="$WORKDIR/defs" --verify=no check-new
+    "$UPDATECTL" update
```

Comment:
This can never work, so the test always fails. sysupdated doesn't know of `--definitions="$WORKDIR/defs" --verify=no` which need to be passed in.

For `--verify=no`, there's already `SYSTEMD_SYSUPDATE_NO_VERIFY=` you can set.

For `--definitions`, you should add an `SYSTEMD_SYSUPDATE_HOST_DEFINITIONS=` env var. Then in `target_get_argument` if we're operating on the host system and that env var is set use it as an argument for `--definitions`

Then you need to add a drop-in for systemd-sysupdated.service that adds `Environment= SYSTEMD_SYSUPDATE_NO_VERIFY=1` and `Environment= SYSTEMD_SYSUPDATE_HOST_DEFINITIONS=$WORKDIR/defs` (making sure to resolve $WORKDIR, ofc). I suggest doing this somewhere near the top of the test file, and calling `systemctl start systemd-sysupdated` there once (rather than doing it in a loop, like you're doing now)

Reply:
Setting the definitions in this way doesn't quite work with `updatectl`.  It first searches for components to use as targets on the default search paths, and so pointing `systemd-sysupdate` at a different set of host definitions doesn't help it find the targets -- it still believes there are none.

Instead I've modified this to set up the configurations in `/run/sysupdate.d`, which (in my opinion) makes more sense anyhow, as this way we're testing closer to how a real system would be set up.